PathfinderAPI/Util/XML/ElementInfo.cs

```diff
@@ -28,60 +45,222 @@ public override string ToString()
     {
+        if(IsText)
+        {
+            writer.WriteString(Content);
+            return;
+        }
         writer.WriteStartElement(Name, "");
         foreach (var attr in Attributes)
             writer.WriteAttributeString(attr.Key, attr.Value);
-        if (Content == null)
-        {
-            foreach (var child in Children)
-                child.WriteToXML(writer);
-        }
-        else
-        {
-            writer.WriteValue(Content);
-        }
+        foreach (var child in Children)
+            child.WriteToXML(writer);
         writer.WriteEndElement();
     }
+
+    public static ElementInfo FromText(string input)
+        => new ElementInfo
+        {
+            Content = input,
+            Type = XmlNodeType.Text,
+            Attributes = null,
+            Children = null
+        };
 }
 
-public static class ListExtensions
+public static class ElementInfoStringExtensions
 {
-    public static ElementInfo GetElement(this List<ElementInfo> list, string elementName)
-    {
-        foreach (var possibleInfo in list)
-        {
-            if (possibleInfo.Name == elementName)
-            {
-                return possibleInfo;
-            }
-        }
+    public static bool TryAsBoolean(this string content, out bool result)
+        => bool.TryParse(content, out result);
+    public static bool TryAsInt(this string content, out int result)
+        => int.TryParse(content, out result);
+    public static bool TryAsFloat(this string content, out float result)
+        => float.TryParse(content, out result);
 
-        return null;
+    public static bool AsBooleanSafe(this string content, bool defaultVal = default)
+        => content.TryAsBoolean(out var value)
+            ? value
+            : defaultVal;
+    public static int AsIntSafe(this string content, int defaultVal = default)
+        => content.TryAsInt(out var value)
+            ? value
+            : defaultVal;
+    public static float AsFloatSafe(this string content, float defaultVal = default)
+        => content.TryAsFloat(out var value)
+            ? value
+            : defaultVal;
+
+    public static bool AsBoolean(this string content, string valName = "content")
+        => content.TryAsBoolean(out var value)
+            ? value
+            : throw new FormatException($"Value of '{valName}' is '{content}' which is not true or false");
+    public static int AsInt(this string content, string valName = "content")
+        => content.TryAsInt(out var value)
+            ? value
+            : throw new FormatException($"Value of '{valName}' is '{content}' which is not an integer, e.g.: 0, 1, 2");
+    public static float AsFloat(this string content, string valName = "content")
+        => content.TryAsFloat(out var value)
+            ? value
+            : throw new FormatException($"Value of '{valName}' is '{content}' which is not a float, e.g.: 1.0");
+}
+
+public static class ElementInfoListExtensions
```

Comment:
What is being done here? These methods are not extension methods in this class, yet the ones in the obsolete class got their API changed (it was changed from List to IEnumerable)

Reply:
This was a screwup, I'll fix it, would've been a breaking change if I don't.